src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs

```diff
@@ -1723,6 +1723,22 @@ private void validate()
                     throw new Exception("Please provide parameter '-SshKeyName' to be used with '-GenerateSshKey'");
                 }
             }
+
+            if (this.IsParameterBound(c => c.VM))
+            {
+                if (this.ParameterSetName == "DefaultParameterSet")
```

Comment:
This check seems redundant because VM is only bound in default parameter set

Reply:
I'll merge this PR. Feel free to raise anther PR to resolve this.